src/torchmetrics/collections.py

```diff
@@ -430,7 +430,9 @@ def persistent(self, mode: bool = True) -> None:
             m.persistent(mode)
 
     def add_metrics(
-        self, metrics: Union[Metric, Sequence[Metric], dict[str, Metric]], *additional_metrics: Metric
+        self,
+        metrics: Union[Metric, Sequence[Metric], dict[str, Metric], "MetricCollection"],
```

Comment:
```suggestion
        metrics: Union[Metric, "MetricCollection", Sequence[Union[Metric, "MetricCollection"]], dict[str, Union[Metric, MetricCollection]]],
```
lets fix the typing, since a metric collection also accepts list/dicts of other metric collections